pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -250,9 +301,25 @@ spec:
                       description: Tags is a map of key/value tags used to select
                         subnets Specifying '*' for a value selects all values for
                         a given tag key.
+                      maxProperties: 30
                       type: object
+                      x-kubernetes-validations:
+                      - message: tags cannot be empty
+                        rule: self.size() != 0
+                      - message: empty tag keys and values aren't supported
+                        rule: self.all(k, k != '' && self[k] != '')
                   type: object
+                maxItems: 20
                 type: array
+                x-kubernetes-validations:
+                - message: securityGroupSelectorTerms cannot be empty
+                  rule: self.size() != 0
```

Comment:
Does this rule apply when the field is ommitted?

Reply:
The field can't be omitted, its required in the CRD

Reply:
Ah yes, you're right :)